### PR TITLE
Update yaml and sync with commonalities

### PR DESCRIPTION
**Pull request by fernandopradocabrillo**
#### What type of PR is this?

Add one of the following kinds:
* correction

#### What this PR does / why we need it:
1. Align securitySchemes with openID as agreed in I&C
2. Format scopes correctly according to Commonalities

#### Which issue(s) this PR fixes:

Fixes (https://github.com/camaraproject/NumberVerification/issues/49) Defines CAMARA scope for the CAMARA version

#### Special notes for reviewers:

N/A

#### Changelog input

N/A

#### Additional documentation 

N/A

### CODE REVIEW

**Review by hdamker — COMMENTED** [review]
I don't think that the X-Correlator header need to be defined within the OAS file. At least that is my understanding of the Design Guidelines and I haven't seen it yet in other sub projects.

**Comment by bigludo7** [comment]
@hdamker As commented in the issue [49](https://github.com/camaraproject/SimSwap/issues/49) we have this `x-correlator` present in[ number verification API](https://github.com/camaraproject/NumberVerification/blob/main/code/API_definitions/number_verification.yaml) . As we have developer implementing both API seems to me important to have some consistency between them.

**Comment by fernandopradocabrillo** [comment]
@hdamker  I agree with you, the guidelines should be less ambiguous on this issue but, as @bigludo7 said, I also think it is important to have consistency and I prefer it to be defined in the OAS file as we have in the different APIs:

- [Home Devices QoD](https://github.com/camaraproject/HomeDevicesQoD/blob/main/code/API_definitions/home_devices_qod.yaml#L82)
- [OTP SMS](https://github.com/camaraproject/OTPvalidationAPI/blob/main/code/API_definitions/one-time-password-sms.yaml#L48)

**Comment by bigludo7** [comment]
@fernandopradocabrillo Probably we have to wait for https://github.com/camaraproject/Commonalities/pull/57 green light before to merge this one - correct?

@hdamker  do you think adding x-Correlator is a blocking issue for this PR?

**Comment by ECORMAC** [comment]
When we state to include OpenID schema what exactly are we proposing (e.g. parts of the standard defined schema, all of it?)

**Comment by fernandopradocabrillo** [comment]
> @fernandopradocabrillo Probably we have to wait for [camaraproject/Commonalities#57](https://github.com/camaraproject/Commonalities/pull/57) green light before to merge this one - correct?

@bigludo7 Yes, also think we have to wait.



> When we state to include OpenID schema what exactly are we proposing (e.g. parts of the standard defined schema, all of it?)

@ECORMAC the proposal is to use OIDC with a 3-legged grant flow. But no mention to further requirements like id_tokens at the moment.

**Comment by hdamker** [comment]
> @hdamker do you think adding x-Correlator is a blocking issue for this PR?

I wouldn't call it blocking, but we could carve it out here until clarified in Commonalities. I still think that it is just cluttering the OAS.

**Comment by DT-DawidWroblewski** [comment]
@fernandopradocabrillo can you please share Number Verification API description that has been used for Local Champion Spain?

Thx!

**Comment by fernandopradocabrillo** [comment]
> @fernandopradocabrillo can you please share Number Verification API description that has been used for Local Champion Spain?
> 
> Thx!

Hi @DT-DawidWroblewski 
That is GSMA related information so I don't think this is the place to ask for something like that. Please, remove the comment and we can discuss it by email.

Thx!

File: code/API_definitions/number_verification.yaml [hunk 1]
```diff
@@ -70,8 +70,8 @@ paths:
       summary: Verifies if the received hashed/plain text phone number matches the phone number associated with the access token
       description: |
         Verifies if the specified phone number (either in plain text or hashed format) matches the one that the user is currently using. Only one of the plain or hashed formats must be provided.
-         - The number verification will be done for the user that has authenticated via mobile network and so their `sub` is in the access token
-         - It returns true/false depending on if the hashed phone number received as input matches the authenticated user's `device phone number` associated to the access token
+        - The number verification will be done for the user that has authenticated via mobile network and so their `sub` is in the access token
```
Comment (DT-DawidWroblewski) [inline code/API_definitions/number_verification.yaml:73]: if we use "OpenID Connect" sub is one of id_token claims.

@fernandopradocabrillo please review

File: code/API_definitions/number_verification.yaml [hunk 2]
```diff
@@ -111,17 +111,17 @@ paths:
...
   /device-phone-number:
     get:
       tags:
         - Phone number share
       summary: Returns the phone number associated with the access token
       description: |
         Returns the phone number so the API clients can verify the number themselves:
-         - It will be done for the user that has authenticated via mobile network and so their `sub` is in the access token
-         - It returns the authenticated user's `device phone number` associated to the access token
+        - It will be done for the user that has authenticated via mobile network and so their `sub` is in the access token
```
Comment (DT-DawidWroblewski) [inline code/API_definitions/number_verification.yaml:123]: idem

File: code/API_definitions/number_verification.yaml (hunk 1 above)
Comment (fernandopradocabrillo) [reply code/API_definitions/number_verification.yaml:73]: afaik access_tokens also contains the `sub`: https://auth0.com/docs/authenticate/login/oidc-conformant-authentication/oidc-adoption-access-tokens#standard-claims

Comment (DT-DawidWroblewski) [reply code/API_definitions/number_verification.yaml:73]: it depends...

following RFC:
```
Access tokens can have different formats, structures, and methods of
   utilization (e.g., cryptographic properties) based on the resource
   server security requirements.  Access token attributes and the
   methods used to access protected resources are beyond the scope of
   this specification and are defined by companion specifications such
   as RFC6750.
```
[RFC6750](https://datatracker.ietf.org/doc/html/rfc6750)
[Access Token](https://datatracker.ietf.org/doc/html/rfc6749#section-1.4)

and sub required claim inside ID Token, that is something specific for OpenId Connect, as it is designed for authentication (& authorization).
[ID Token](https://openid.net/specs/openid-connect-core-1_0.html#IDToken)

So you can state that:

sub is **build in OpenID Connect**, as it is a part of id_token, and is **optional for OAUTH2.0**  as it depends on _resource server security requirements_.

Therefore, if we're willing to use _access_tokens_ we should use OAUTH2.0 security scheme.

**Comment by ECORMAC** [comment]
Agree with David, "sub" is defined in relation to identity tokens (can't find it being mentioned in any spec in relation to access tokens). The current description in the Yaml is confusing, it should be clarified what we are proposing.

**Comment by fernandopradocabrillo** [comment]
@DT-DawidWroblewski 
As discussed in the meeting I have removed references to the `sub` in the access token

**Review by DT-DawidWroblewski — APPROVED** [review]
LGTM

**Pull request merged**